src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -157,6 +202,11 @@ public function merge(AttributeMetadataInterface $attributeMetadata)
             $this->serializedName = $attributeMetadata->getSerializedName();
         }
 
+        // Overwrite only if empty or nullable array
+        if ([] == $this->serializedNames || null == $this->serializedNames) {
```

Comment:
```suggestion
        if ($this->serializedNames) {
```

Reply:
keep the condition sense by adding `!` after suggestion